GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -349,37 +419,91 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            foreach (var pos in positions)
             {
-                for (var i = 0; i < positions.Count; ++i)
+                uint inhibitorId = 0;
+                var waypoints = new List<Vector2>();
+                switch (pos)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_MELEE, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    case MinionSpawnPosition.SPAWN_BLUE_BOT:
+                        waypoints = _blueBotWaypoints;
+                        inhibitorId = 0xff26ac0f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_MID:
+                        waypoints = _blueMidWaypoints;
+                        inhibitorId = 0xffff8f1f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_TOP:
+                        waypoints = _blueTopWaypoints;
+                        inhibitorId = 0xff6793d0;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_BOT:
+                        waypoints = _redBotWaypoints;
+                        inhibitorId = 0xff9303e1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_MID:
+                        waypoints = _redMidWaypoints;
+                        inhibitorId = 0xff4a20f1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_TOP:
+                        waypoints = _redTopWaypoints;
+                        inhibitorId = 0xffd23c3e;
+                        break;
                 }
-                return false;
-            }
+                var inhibitorIsDead = (GetInhibitorById(inhibitorId).getState() == InhibitorState.Dead);
 
-            if (_waveNumber == 3)
-            {
-                for (var i = 0; i < positions.Count; ++i)
+                if (_gameTime > 35 * 60 * 1000)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_CANNON, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    var list = inhibitorIsDead ? _superCannonMinionWave : _cannonMinionWave;
```

Comment:
I would avoid the `condition ? option1 : option2` syntax sugar, as it tends to make things harder to read. Personally I'd use
```CS
var list = _cannonMinionWave;
if(inhibitorIsDead)
{
    list = _superCannonMinionWave;
}
```

You don't have to change this if you don't want to though, as it's only one condition and fairly clear.